SmartDeviceLink/SDLCreateWindow.h

```diff
@@ -0,0 +1,113 @@
+ 
+ Widgets can be associated with a specific App Service type such as `MEDIA` or `NAVIGATION`. As an example if a `MEDIA` app becomes active, this app becomes audible and is allowed to play audio. If the media app has created a widget with `MEDIA` type associated, this widget will automatically be activated together with the app.
+ 
+ @since SDL 6.0
+ */
+@interface SDLCreateWindow : SDLRPCRequest
+
+
+/**
+ Constructor with the required parameters
+
+ @param windowId The type of the window to be created. Main window or widget.
+ @param windowName The window name to be used by the HMI. The name of the pre-created default window will match the app name.
+ Multiple apps can share the same window name except for the default main window.
+ Creating a window with a name which is already in use by the app will result in `DUPLICATE_NAME`.
+ MaxLength 100.
+ @param windowType The type of the window to be created. Main window or widget.
+ */
+- (instancetype)initWithId:(NSInteger)windowId windowName:(NSString *)windowName windowType:(SDLWindowType)windowType;
+
+/**
+ Convinience constructor with nullable optional parameters.
+ 
+ @param windowId The type of the window to be created. Main window or widget.
+ @param windowName The window name to be used by the HMI. The name of the pre-created default window will match the app name.
+ Multiple apps can share the same window name except for the default main window.
+ Creating a window with a name which is already in use by the app will result in `DUPLICATE_NAME`.
+ MaxLength 100.
+ @param windowType The type of the window to be created. Main window or widget.
+ @param associatedServiceType Allows an app to create a widget related to a specific service type.
+ As an example if a `MEDIA` app becomes active, this app becomes audible and is allowed to play audio. Actions such as skip or play/pause will be
+ directed to this active media app. In case of widgets, the system can provide a single "media" widget which will act as a placeholder for the active media app.
+ 
+ It is only allowed to have one window per service type. This means that a media app can only have a single MEDIA widget. Still the app can create widgets omitting this parameter. Those widgets would be available as app specific widgets that are permanently included in the HMI.
+ 
+ This parameter is related to widgets only. The default main window, which is pre-created during app registration, will be created based on the HMI types specified in the app registration request.
+ */
+- (instancetype)initWithId:(UInt32)windowId windowName:(NSString *)windowName windowType:(SDLWindowType)windowType associatedServiceType:(nullable NSString *)associatedServiceType;
```

Comment:
correct but I put it because there is an optional Int

Reply:
thanks @joeljfischer I was kinda folling this conversation: https://github.com/smartdevicelink/sdl_ios/pull/1350#discussion_r316335424
Maybe I miss understood something. But is it ok to create it like below even if it is not as an option because it is a uint32? 
`- (instancetype)initWithId:(NSInteger)windowId windowName:(NSString *)windowName windowType:(SDLWindowType)windowType associatedServiceType:(nullable NSString *)associatedServiceType duplicateUpdatesFromWindowID:(UInt32)duplicateUpdatesFromWindowID;`